homeassistant/components/enocean/cover.py

```diff
@@ -0,0 +1,267 @@
+    }
+)
+
+
+def setup_platform(
+    hass: HomeAssistant,
+    config: ConfigType,
+    add_entities: AddEntitiesCallback,
+    discovery_info: DiscoveryInfoType | None = None,
+) -> None:
+    """Set up the Cover platform for EnOcean."""
+    dev_id = config.get(CONF_ID)
+    sender_id = config.get(CONF_SENDER_ID)
+    dev_name = config.get(CONF_NAME)
+    device_class = config.get(CONF_DEVICE_CLASS)
+    if device_class is None:
+        device_class = DEVICE_CLASS_BLIND
+    watchdog_timeout = config.get(WATCHDOG_TIMEOUT)
+    add_entities(
+        [
+            EnOceanCover(
+                sender_id, dev_id, dev_name, device_class, hass, watchdog_timeout
+            )
+        ]
+    )
+
+
+class EnOceanCover(EnOceanEntity, CoverEntity):
+    """Representation of an EnOcean Cover (EEP D2-05-00)."""
+
+    def __init__(
+        self, sender_id, dev_id, dev_name, device_class, hass, watchdog_timeout
+    ):
+        """Initialize the EnOcean Cover."""
+        super().__init__(dev_id, dev_name)
+        self._device_class = device_class
+        self._position = None
+        self._is_closed = None
+        self._is_opening = False
+        self._is_closing = False
+        self._sender_id = sender_id
+        self._dev_name = dev_name
+        self._attr_unique_id = f"{combine_hex(dev_id)}-{device_class}"
+        self._state_changed_by_command = False
+        self._watchdog_enabled = False
+        self._watchdog_seconds_remaining = 0
+        self._watchdog_timeout = watchdog_timeout
+        self._hass = hass
+
+    @property
+    def name(self):
+        """Return the default name for the cover."""
+        return self._dev_name
+
+    @property
+    def device_class(self):
+        """Return the class of this cover."""
+        return self._device_class
+
+    @property
+    def supported_features(self) -> int:
+        """Flag supported features."""
+        flags = (
+            CoverEntityFeature.OPEN
+            | CoverEntityFeature.CLOSE
+            | CoverEntityFeature.STOP
+            | CoverEntityFeature.SET_POSITION
+        )
+        return flags
+
+    @property
+    def current_cover_position(self) -> int | None:
+        """Return the current cover position."""
+        return self._position
+
+    @property
+    def is_opening(self) -> bool | None:
+        """Return if the cover is opening or not."""
+        return self._is_opening
+
+    @property
+    def is_closing(self) -> bool | None:
+        """Return if the cover is closing or not."""
+        return self._is_closing
+
+    @property
+    def is_closed(self) -> bool | None:
+        """Return if the cover is closed or not."""
+        return self._is_closed
+
+    def open_cover(self, **kwargs) -> None:
+        """Open the cover."""
+        self._state_changed_by_command = True
+        self._is_opening = True
+        self._is_closing = False
+        self.start_or_feed_watchdog()
+
+        telegram = [0xD2, 0, 0, 0, 1]
```

Comment:
This is protocol details that should be part of a 3rd party library.

Reply:
Agreed. Fortunately the used EnOcean-library already has better high-level functions, please have a look at commit [3dd3a08](https://github.com/home-assistant/core/pull/75356/commits/3dd3a085cc5c62c775f5783881ba0faccb6c4836).